Let EditorPlugins set editor/distraction-free mode

willnationsdev opened the pull request:
Give plugin creators the ability to manipulate which main screen editor is displayed and whether the editor takes up the window's space or not. This also applies to a plugin's custom editor.

willnationsdev committed:
Let EditorPlugins set editor/distraction-free mode

akien-mga commented on editor/editor_node.cpp:
```diff
@@ -2775,6 +2775,16 @@ void EditorNode::_editor_select(int p_which) {
 }
 
+void EditorNode::select_editor_by_name(const String &p_name) {
+	if ("" == p_name)
```
We write conditions the other way around in general, `p_name == ""`.

Should it print an error if no editor is found that matches `p_name`?

akien-mga commented on editor/editor_plugin.cpp:
```diff
@@ -288,6 +296,9 @@ void EditorInterface::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("save_scene"), &EditorInterface::save_scene);
 	ClassDB::bind_method(D_METHOD("save_scene_as", "path", "with_preview"), &EditorInterface::save_scene_as, DEFVAL(true));
+
+	ClassDB::bind_method(D_METHOD("set_main_screen_editor_by_name", "name"), &EditorInterface::set_main_screen_editor_by_name);
```
I don't know if the `_by_name` suffix is necessary since the argument is already a String called `name`, so it's relatively explicit.

willnationsdev commented:
I have swapped the `p_name == ""` check around and added ERR_FAIL commands in to produce visible errors in the erroneous use cases. I also updated the name in the EditorPlugin/scripting API to not use the `_by_name` suffix. However, the EditorNode has an `_editor_select()` method that accepts an index into the list of toolbuttons at the top. Since plugins don't know where they fall within that list, I thought it better to let them switch contexts based on the name of the context/editor rather than its position in the list, hence the addition of the `_by_name` suffix for the EditorNode method.

Let me know what you think.
